src/test/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStepTest.java

```diff
@@ -186,16 +197,26 @@ public InvalidChangelogSCM() { }
         @Override public void checkout(Run<?,?> build, Launcher launcher, FilePath workspace, TaskListener listener, File changelogFile, SCMRevisionState baseline) throws IOException, InterruptedException {
             // Unlike the superclass, which adds an XML header to the file, we just ignore the file.
         }
-        @Override public ChangeLogParser createChangeLogParser() {
-            return new ChangeLogParser() {
-                public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
-                    Jenkins.XSTREAM2.fromXML(changelogFile);
-                    throw new AssertionError("The previous line should always fail because the file is empty");
-                }
-            };
-        }
+        ChangeLogParser changeLogParser = createChangeLogParser();
         @TestExtension("scmParsesUnmodifiedChangelogFile")
         public static class DescriptorImpl extends NullSCM.DescriptorImpl { }
     }
 
+    public static class FakeChangeLogSCM extends NullSCM {
+        @DataBoundConstructor
+        public FakeChangeLogSCM() { }
+        ChangeLogParser changeLogParser = createChangeLogParser();
+        @TestExtension("scmParsesFakeChangeLogSCMChangelogFile")
+        public static class DescriptorImpl extends NullSCM.DescriptorImpl { }
+    }
+
+    public ChangeLogParser createChangeLogParser() {
+        return new ChangeLogParser() {
+            @Override public ChangeLogSet<? extends ChangeLogSet.Entry> parse(Run build, RepositoryBrowser<?> browser, File changelogFile) throws IOException, SAXException {
```

Comment:
Unclear if I need this `@Override` or not. The test works fine with or without it.

Reply:
It seems like `@Override` it is optional but basically everyone uses it.

> Using the @Override annotation acts as a compile-time safeguard against a common programming mistake. 

https://stackoverflow.com/questions/94361/when-do-you-use-javas-override-annotation-and-why